examples/transactions.c

```diff
@@ -184,135 +154,137 @@ create_transactional_producer (const rd_kafka_topic_partition_t *rktpar) {
+        err = rd_kafka_assignment(consumer, &offsets);
+        if (err) {
+                fprintf(stderr, "No current assignment to rewind: %s\n",
+                        rd_kafka_err2str(err));
+                return;
+        }
 
-        /* Abort the current transaction, ignore any errors
-         * since we're terminating the producer anyway. */
-        error = rd_kafka_abort_transaction(rk, -1);
-        if (error) {
-                fprintf(stderr,
-                        "WARNING: Ignoring abort_transaction() error since "
-                        "producer is being destroyed: %s\n",
-                        rd_kafka_error_string(error));
-                rd_kafka_error_destroy(error);
+        if (offsets->cnt == 0) {
+                fprintf(stderr, "No current assignment to rewind\n");
+                rd_kafka_topic_partition_list_destroy(offsets);
+                return;
         }
 
-        rd_kafka_destroy(rk);
-}
+        /* Note: Timeout must be lower than max.poll.interval.ms */
+        err = rd_kafka_committed(consumer, offsets, 10*1000);
+        if (err)
+                fatal("Failed to acquire committed offsets: %s",
+                      rd_kafka_err2str(err));
+
+        /* Seek to committed offset, or start of partition if no
+         * committed offset is available. */
+        for (i = 0 ; i < offsets->cnt ; i++) {
+                /* No committed offset, start from beginning */
+                if (offsets->elems[0].offset < 0)
+                        offsets->elems[0].offset =
+                                RD_KAFKA_OFFSET_BEGINNING;
+        }
 
+        /* Perform seek */
+        error = rd_kafka_seek_partitions(consumer, offsets, -1);
+        if (error)
+                fatal_error("Failed to seek", error);
 
+        rd_kafka_topic_partition_list_destroy(offsets);
+}
 
 /**
  * @brief Abort the current transaction and rewind consumer offsets to
  *        position where the transaction last started, i.e., the committed
- *        consumer offset.
+ *        consumer offset, then begin a new transaction.
  */
-static void abort_transaction_and_rewind (struct state *state) {
-        rd_kafka_topic_t *rkt = rd_kafka_topic_new(consumer,
-                                                   state->rktpar->topic, NULL);
-        rd_kafka_topic_partition_list_t *offset;
-        rd_kafka_resp_err_t err;
+static void abort_transaction_and_rewind (rd_kafka_t *consumer,
+                                          rd_kafka_t *producer) {
         rd_kafka_error_t *error;
 
-        fprintf(stdout,
-                "Aborting transaction and rewinding offset for %s [%d]\n",
-                state->rktpar->topic, state->rktpar->partition);
+        fprintf(stdout, "Aborting transaction and rewinding offsets\n");
 
         /* Abort the current transaction */
-        error = rd_kafka_abort_transaction(state->producer, -1);
+        error = rd_kafka_abort_transaction(producer, -1);
```

Comment:
might as well throw in an unactionable complaint here about there being no mechanism for the application to abort an operation with infinite timeout.

Reply:
`rd_kafka_queue_yield()` is half-way there..